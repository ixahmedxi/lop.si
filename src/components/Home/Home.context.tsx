import { createContext, Dispatch, SetStateAction, useState } from 'react'

interface IHomeContext {
  id: string
  setId: Dispatch<SetStateAction<string>>
}

export const HomeContext = createContext<IHomeContext>({
  id: '',
  setId: () => null
})

export const HomeProvider: React.FC = ({ children }) => {
  const [id, setId] = useState('')
  return <HomeContext.Provider value={{ id, setId }}>{children}</HomeContext.Provider>
}
