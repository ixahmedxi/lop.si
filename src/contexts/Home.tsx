import { createContext, Dispatch, SetStateAction, useContext, useState } from 'react'

interface IHomeContext {
  id: string
  setId: Dispatch<SetStateAction<string>>
  isLoading: boolean
  setIsLoading: Dispatch<SetStateAction<boolean>>
}

export const HomeContext = createContext<IHomeContext>({
  id: '',
  setId: () => null,
  isLoading: false,
  setIsLoading: () => false
})

export const HomeProvider: React.FC = ({ children }) => {
  const [id, setId] = useState('')
  const [isLoading, setIsLoading] = useState(false)
  return (
    <HomeContext.Provider value={{ id, setId, isLoading, setIsLoading }}>
      {children}
    </HomeContext.Provider>
  )
}

export const useHomeContext = (): IHomeContext => useContext(HomeContext)
