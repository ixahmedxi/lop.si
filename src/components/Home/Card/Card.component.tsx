/* @jsx jsx */
import { useHomeContext } from '@contexts/Home'
import { useNeuBoxShadow } from '@hooks/useBoxShadow'
import Link from 'next/link'
import { FiCheckSquare, FiClipboard } from 'react-icons/fi'
import useClipboard from 'react-use-clipboard'
import { Box, Flex, jsx, Spinner, SxStyleProp, Text } from 'theme-ui'

export const Card: React.FC = () => {
  const { id, isLoading } = useHomeContext()
  const cardShadow = useNeuBoxShadow(10, 20)
  const iconButtonShadow = useNeuBoxShadow(3, 6)
  const [isCopied, setCopied] = useClipboard('https://lop.si/' + id, { successDuration: 2000 })

  const styles: SxStyleProp = {
    column: {
      width: [`${1000 / 12}%`, `${800 / 12}%`, `${600 / 12}%`, `${400 / 12}%`],
      borderRadius: 20,
      overflow: 'hidden',
      mt: 11,
      opacity: id === '' ? 0 : 1,
      mx: 'auto',
      minHeight: '200px',
      transition: '0.2s ease-out',
      position: 'relative',
      ...cardShadow
    },
    cardTitle: { fontWeight: 'heading', fontSize: 1, color: 'primary' },
    cardDescription: { fontSize: 0, pt: 1, lineHeight: 1, opacity: 0.7 },
    linkRow: {
      pt: [1, 1, 2],
      alignItems: 'center',
      justifyContent: 'space-between'
    },
    iconButton: {
      ...iconButtonShadow,
      width: 5,
      height: 5,
      borderRadius: 50,
      border: 'none',
      bg: 'background',
      color: 'primary',
      cursor: 'pointer',
      outline: 'none'
    },
    copiedBox: {
      width: '100%',
      bg: 'primary',
      color: 'background',
      borderTopLeftRadius: 20,
      borderTopRightRadius: 20,
      textAlign: 'center',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
      height: isCopied ? 5 : 0,
      transition: '0.2s ease-out'
    }
  }

  return (
    <Box sx={styles.column}>
      <Box sx={styles.copiedBox}>
        <Text
          as="p"
          sx={{
            display: 'flex',
            alignItems: 'center',
            justifyContent: 'center',
            opacity: isCopied ? 1 : 0
          }}
        >
          <FiCheckSquare sx={{ mr: 1 }} /> Copied to clipboard!
        </Text>
      </Box>
      <Box
        sx={{
          p: [4, 4, 5],
          opacity: isLoading ? 0 : 1,
          transition: 'opacity 0.2s ease-out'
        }}
      >
        <Text as="h2" sx={styles.cardTitle}>
          Your created short url:
        </Text>
        <Text as="p" sx={styles.cardDescription}>
          Here&#39;s your newly created short url for you to freely use everywhere.
        </Text>
        <Flex sx={styles.linkRow}>
          <Link href={'/' + id}>
            <a
              sx={{
                color: 'primary',
                fontSize: 1,
                textDecoration: 'underline',
                cursor: 'pointer'
              }}
            >
              {'lop.si/' + id}
            </a>
          </Link>
          <button
            type="button"
            sx={styles.iconButton}
            onClick={setCopied}
            aria-label="copy to clipboard"
          >
            <FiClipboard />
          </button>
        </Flex>
      </Box>
      <Flex
        sx={{
          borderRadius: 20,
          height: '100%',
          width: '100%',
          position: 'absolute',
          left: 0,
          top: 0,
          alignItems: 'center',
          justifyContent: 'center',
          bg: 'background',
          opacity: !isLoading ? 0 : 1,
          zIndex: !isLoading ? -1 : 1,
          transition: 'opacity 0.2s ease-out'
        }}
      >
        <Spinner sx={{ color: 'primary' }} size={48} />
      </Flex>
    </Box>
  )
}
