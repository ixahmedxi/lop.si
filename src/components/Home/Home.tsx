import { HomeProvider } from '@contexts/Home'
import { NextPage } from 'next'
import { Card } from './Card/Card.component'
import { Form } from './Form/Form.component'
import { Introduction } from './Introduction/Introduction.component'

const Home: NextPage = () => {
  return (
    <HomeProvider>
      <Introduction />
      <Form />
      <Card />
    </HomeProvider>
  )
}

export default Home
