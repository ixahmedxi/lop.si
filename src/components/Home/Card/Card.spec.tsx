import { HomeProvider } from '@contexts/Home'
import { render } from 'test-utils'
import { Card } from './Card.component'

describe('Card component', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <HomeProvider>
        <Card />
      </HomeProvider>
    )

    expect(container).toMatchSnapshot()
  })
})
