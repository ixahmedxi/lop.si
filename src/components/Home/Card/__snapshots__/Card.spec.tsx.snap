// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component matches snapshot 1`] = `
@keyframes animation-0 {
  from {
    -webkit-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  to {
    -webkit-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.emotion-12 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 83.33333333333333%;
  border-radius: 20px;
  overflow: hidden;
  margin-top: 88px;
  opacity: 0;
  margin-left: auto;
  margin-right: auto;
  min-height: 200px;
  -webkit-transition: none;
  transition: none;
  position: relative;
  box-shadow: none;
}

@media screen and (min-width:576px) {
  .emotion-12 {
    width: 66.66666666666667%;
  }
}

@media screen and (min-width:768px) {
  .emotion-12 {
    width: 50%;
  }
}

@media screen and (min-width:992px) {
  .emotion-12 {
    width: 33.333333333333336%;
  }
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 100%;
  background-color: var(--theme-ui-colors-primary,#8093f2);
  color: var(--theme-ui-colors-background,#2B2D32);
  border-top-left-radius: 20px;
  border-top-right-radius: 20px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 0;
  -webkit-transition: 0.2s ease-out;
  transition: 0.2s ease-out;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  opacity: 0;
}

.emotion-0 {
  margin-right: 8px;
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 32px;
  opacity: 1;
  -webkit-transition: opacity 0.2s ease-out;
  transition: opacity 0.2s ease-out;
}

@media screen and (min-width:576px) {
  .emotion-8 {
    padding: 32px;
  }
}

@media screen and (min-width:768px) {
  .emotion-8 {
    padding: 40px;
  }
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-weight: 700;
  font-size: 16px;
  color: var(--theme-ui-colors-primary,#8093f2);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 12px;
  padding-top: 8px;
  line-height: 1.8;
  opacity: 0.7;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media screen and (min-width:576px) {
  .emotion-7 {
    padding-top: 8px;
  }
}

@media screen and (min-width:768px) {
  .emotion-7 {
    padding-top: 16px;
  }
}

.emotion-5 {
  color: var(--theme-ui-colors-primary,#8093f2);
  font-size: 16px;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

.emotion-6 {
  -webkit-transition: none;
  transition: none;
  box-shadow: none;
  width: 40px;
  height: 40px;
  border-radius: 50px;
  border: none;
  background-color: transparent;
  color: var(--theme-ui-colors-primary,#8093f2);
  cursor: pointer;
  outline: none;
}

.emotion-11 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-radius: 20px;
  height: 100%;
  width: 100%;
  position: absolute;
  left: 0;
  top: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: var(--theme-ui-colors-background,#2B2D32);
  opacity: 0;
  z-index: -1;
  -webkit-transition: opacity 0.2s ease-out;
  transition: opacity 0.2s ease-out;
}

.emotion-10 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-primary,#8093f2);
  overflow: visible;
  color: var(--theme-ui-colors-primary,#8093f2);
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-transform-origin: 50% 50%;
  -ms-transform-origin: 50% 50%;
  transform-origin: 50% 50%;
  -webkit-animation-name: animation-0;
  animation-name: animation-0;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-duration: 500ms;
  animation-duration: 500ms;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

<div>
  <div
    class="emotion-12"
    data-testid="card"
  >
    <div
      class="emotion-2"
    >
      <p
        class="emotion-1"
      >
        <svg
          class="emotion-0"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 11 12 14 22 4"
          />
          <path
            d="M21 12v7a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h11"
          />
        </svg>
         Copied to clipboard!
      </p>
    </div>
    <div
      class="emotion-8"
    >
      <h2
        class="emotion-3"
      >
        Your created short url:
      </h2>
      <p
        class="emotion-4"
      >
        Here's your newly created short url for you to freely use everywhere.
      </p>
      <div
        class="emotion-7"
      >
        <a
          class="emotion-5"
        >
          lop.si/
        </a>
        <button
          aria-label="copy to clipboard"
          class="emotion-6"
          type="button"
        >
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 4h2a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H6a2 2 0 0 1-2-2V6a2 2 0 0 1 2-2h2"
            />
            <rect
              height="4"
              rx="1"
              ry="1"
              width="8"
              x="8"
              y="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="emotion-11"
    >
      <svg
        class="emotion-10"
        fill="none"
        height="48"
        role="img"
        stroke="currentcolor"
        stroke-width="4"
        viewBox="0 0 32 32"
        width="48"
      >
        <title>
          Loading...
        </title>
        <circle
          cx="16"
          cy="16"
          opacity="0.125"
          r="12"
        />
        <circle
          class="emotion-9"
          cx="16"
          cy="16"
          r="12"
          stroke-dasharray="75.39822368615503"
          stroke-dashoffset="56.548667764616276"
        />
      </svg>
    </div>
  </div>
</div>
`;
